cockatrice/src/deckview.cpp

```diff
@@ -116,6 +117,34 @@ void DeckViewCard::mouseMoveEvent(QGraphicsSceneMouseEvent *event)
     setCursor(Qt::OpenHandCursor);
 }
 
+void DeckView::mouseDoubleClickEvent(QMouseEvent *event)
+{
+    if (event->button() == Qt::LeftButton)
+    {
+        qDebug() << "LEFT";
+        QList<MoveCard_ToZone> *result = new QList<MoveCard_ToZone>();
+        QList<QGraphicsItem *> sel = scene()->selectedItems();
+
+        for (int i = 0; i < sel.size(); i++) {
+            DeckViewCard *c = static_cast<DeckViewCard *>(sel.at(i));
+            MoveCard_ToZone m;
+            m.set_card_name(c->getName().toStdString());
+            m.set_start_zone(c->getOriginZone().toStdString());
+
+            if (c->getOriginZone() == "main")
```

Comment:
The main problem now is that you are using `getOriginZone()` to determine where a card actually is and where it should move to, but the "original zone" is never updated, so if you move a card from "main" to "side" its original zone will still be "main"; double clicking it again will try to move it again from the main to the side.
You don't want to che the card "original zone", but its "current zone" instead; you should be able to get this information from the card's [parentItem()](http://doc.qt.io/qt-5.8/qgraphicsitem.html#parentItem)